### PR TITLE
Added custom folder and fragment ion support

### PR DESCRIPTION
**Pull request by maddyboo**
Hi,

I found MScheck via your recent ChemRxiv publication (https://chemrxiv.org/engage/chemrxiv/article-details/641ef81e647e3dca997c2ba6) and have found this to be a really useful tool!

In using it, I found a few features I wanted to add, and so have done so in this branch:

- Added the choice of a custom output folder
- Added the functionality to search for fragmentation ions (e.g. the commonly observed McLafferty Boc fragmentation - resulting in -tBu and -Boc masses showing up). This also resulted in a minor change to the report title above the found MS ion chart(s)
- I also added a fix to close matplotlib figures after use (in our use case we are using MScheck in a loop for plate analysis, once the no. of figures exceeds 20 (i.e. ~20 analyses) a memory warning would show up in the terminal although the process would still complete).

Thanks for sharing this tool! 

Daniel

### CODE REVIEW

File: mscheck/analyse.py [hunk 1]
```diff
@@ -161,7 +164,18 @@ def analyse(self, compoundsmiles: str, ionstoadd: list, tolerance: int) -> dict:
         ions_to_add_mols = [get_mol(ion) for ion in ionstoadd]
         ions_to_add_MW = [get_MW(mol) for mol in ions_to_add_mols]
 
-        for ion_to_add_mol, MW in zip(ions_to_add_mols, ions_to_add_MW):
+        if ionstosub:
+            ions_to_sub_mols = [get_mol(ion) for ion in ionstosub]
+            # Mass +1 to account for resultant +H adduct observed - assuming positive ionisation method
+            ions_to_sub_MW = [1 - get_MW(mol) for mol in ions_to_sub_mols]
```
Comment (Waztom) [inline mscheck/analyse.py:170]: @maddyboo should this be [1 + get_MW....] to add a proton to the adduct?

Comment (Waztom) [reply mscheck/analyse.py:170]: @maddyboo nice addition btw and thank you!

File: mscheck/analyse.py [hunk 2]
```diff
@@ -219,10 +233,13 @@ def get_strongest_mz_pattern(self, mz_values: list) -> largest_mz_pattern:
         mz_intensities_max = mz_intensities[max_index]
```
Comment (Waztom) [inline mscheck/analyse.py:233]: @maddyboo set reports to a default value in the function

**Review by Waztom — COMMENTED** [review]
@maddyboo this is awesome, and thank you for contributing! Please double-check/confirm the +1 or -1 proton to adduct bit, but otherwise good to go!

**Review by maddyboo — COMMENTED** [review]
Thanks for double checking! Please find a more verbose explanation in the comments!

**Pull request merged**